lib/signs/utilities/early_am_suppression.ex

```diff
@@ -0,0 +1,308 @@
+          #   do_early_am_suppression since we reset the zone to "m" and the generic paging
+          #   will be reconstructed
+          {%Message.Headways.Paging{
+             destination: headway_top.destination,
+             range: headway_bottom.range
+           }, %{prediction | zone: "m"}}
+
+        _ ->
+          messages
+      end
+
+    {top_status, bottom_status} = statuses
+
+    Enum.map(
+      [
+        {top_message, top_scheduled, top_status},
+        {bottom_message, bottom_scheduled, bottom_status}
+      ],
+      fn {message, scheduled, status} ->
+        if(status == :none,
+          do: {message, %Message.Empty{}},
+          else: get_early_am_content(sign, {message}, scheduled, status, current_time)
+        )
+        |> case do
+          # If line has status :none, then it could be a paging headway message
+          {%Headways.Paging{destination: destination, range: range}, _} ->
+            {%Headways.Top{destination: destination, vehicle_type: :train},
+             %Headways.Bottom{range: range}}
+
+          content ->
+            content
+        end
+      end
+    )
+    |> List.to_tuple()
+  end
+
+  defp get_early_am_content(
+         sign,
+         messages,
+         {scheduled, destination},
+         status,
+         current_time
+       ) do
+    cond do
+      status == :fully_suppressed ->
+        {%EarlyAm.DestinationTrain{destination: destination},
+         %EarlyAm.ScheduledTime{
+           scheduled_time: scheduled
+         }}
+
+      status == :partially_suppressed ->
+        case filter_early_am_messages(messages, sign.id) do
+          [] ->
+            # If no valid predictions, try fetching headways
+            case Signs.Utilities.Headways.get_messages(sign, current_time) do
+              # If no headways are returned, default to timestamp message
+              {%Message.Empty{}, %Message.Empty{}} ->
+                {%EarlyAm.DestinationTrain{
+                   destination: destination
+                 },
+                 %EarlyAm.ScheduledTime{
+                   scheduled_time: scheduled
+                 }}
+
+              {headway_top, headway_bottom} ->
+                # Make sure routes is nil so that destination is used as headsign
+                {%{headway_top | destination: destination, routes: nil}, headway_bottom}
+            end
+
+          [message] ->
+            {message, %Content.Message.Empty{}}
+
+          messages ->
+            List.to_tuple(messages)
+        end
+    end
+  end
+
+  defp filter_early_am_messages(messages, sign_id) do
+    Tuple.to_list(messages)
+    |> Enum.reject(fn
+      message ->
+        match?(%Headways.Top{}, message) or match?(%Headways.Bottom{}, message) or
+          (sign_id not in ["symphony_eastbound", "prudential_eastbound"] and
+             match?(%Message.Predictions{}, message) and message.certainty > 120)
```

Comment:
Note that we might get `StoppedTrain` messages here too, which would currently pass this filter regardless of the underlying certainty value. That may be fine, but I wanted to call it out.